test/compiler/compiler.jl

```diff
@@ -1615,6 +1615,15 @@ g26172(::Val{0}) = ()
 g26172(v) = (nothing, g26172(f26172(v))...)
 @test @inferred(g26172(Val(10))) === ntuple(_ -> nothing, 10)
 
+function conflicting_assignment_conditional()
+    x = iterate([])
+    if x === (x = 4; nothing)
+        return x
+    end
+    return 5
+end
```

Comment:
This is such a beautifully perverse function :) <3

Reply:
`@test_programmer_is_slapped_on_his_fingers_by_robotic_hand`